### ISSUE TITLE
Style Variations do not include all differences. 

### ISSUE THREAD

**Issue by johnstonphilip** [WordPress/gutenberg#42786]
### Description

When I have a style variation in my theme that has a difference like `Allow users to set background colors.`, it doesn't allow me to select that style in the UI. 

### Step-by-step reproduction instructions

1. Create a theme.json file in your theme that looks like this (reduced for simplicity to illustrate the issue):
```
{
    "version": 2,
    "settings": {
        "color": {
            "background":true
        }
    }
}
```
2. In your theme's `styles` directory, create one called "`no-background.json`" that looks like this:
```
{
    "version": 2,
    "settings": {
        "color": {
            "background":false
        }
    }
}
```
3. In the Site Editor UI for Global Styles, notice you cannot choose the one called `no-backgrounds`. Clicking on it has no affect.
<img width="285" alt="Screen Shot 2022-07-28 at 2 07 38 PM" src="https://user-images.githubusercontent.com/7538525/181607371-31410ff6-ec76-42b5-b505-f37eba2a0f4f.png">

### Environment info

- Gutenberg version 13.8.0-rc.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by johnstonphilip** [comment]
Digging to find out why. Possible clue here:
https://github.com/WordPress/gutenberg/blob/33d84b036592a5bf31af05b7710f3b2b14163dc4/packages/edit-site/src/components/global-styles/screen-style-variations.js#L29

**Comment by carolinan** [comment]
Is it a valid style variation if there are no **style** variations only settings? 🤔 
If it is expected, then I guess, yes 🤷‍♀️

**Comment by johnstonphilip** [comment]
Right. If anything from theme.json is allowed in a style variation, than this is a bug.

On the other hand, if there is any value in theme.json that should _not_ be used in a style variation, then there should be a separate [schema](https://github.com/WordPress/gutenberg/tree/trunk/schemas/json) file for style variations so we know what is and is-not allowed.
